src/NuGet.Core/NuGet.Commands/CommandRunners/PushRunner.cs

```diff
@@ -37,8 +38,16 @@ public static class PushRunner
             {
                 timeoutSeconds = 5 * 60;
             }
+            PackageSource packageSource = CommandRunnerUtility.GetOrCreatePackageSource(sourceProvider, source);
+            var packageUpdateResource = await CommandRunnerUtility.GetPackageUpdateResource(sourceProvider, packageSource);
+
+            // Only warn for V3 style sources because they have a service index which is different from the final push url.
+            if (packageSource.IsHttp && !packageSource.IsHttps &&
+                (packageSource.ProtocolVersion == 3 || packageSource.Source.EndsWith("json", StringComparison.OrdinalIgnoreCase)))
+            {
+                logger.LogWarning(string.Format(CultureInfo.CurrentCulture, Strings.Warning_HttpServerUsage, "push", packageSource.Source));
```

Comment:
We are. The check is done inside. 
The reason why we're special casing V3 here is because V3 is the only source that redirects. For the other ones you have to pass the exact target url. 

PushCommand_WhenPushingToAnHttpServerWithSymbols_Warns tests that.